lib/plugins/unthreaded-signing/plugin.c

```diff
@@ -36,35 +37,81 @@
 #endif
 #endif
 
+/**
+ * Structure to store the signature information.
+ */
+typedef struct _sv_unthreaded_signature_info_t {
+  uint8_t *signature;  // The signature of the |hash|.
+  size_t signature_size;  // The size of the |signature|.
+  size_t max_signature_size;  // The allocated size of the |signature|.
+} sv_unthreaded_signature_info_t;
+
 // Plugin handle to store the signature, etc.
 typedef struct _sv_unthreaded_plugin_t {
   bool signature_generated;
+  int signature_data_buffer_idx;
   sign_or_verify_data_t sign_data;
+  sv_unthreaded_signature_info_t
+      signature_data_buffer[MAX_BUFFER_LENGTH];  // Buffer to store signature information
 } sv_unthreaded_plugin_t;
 
+/**
+ * Shifts the elements in the signature data buffer to the left.
+ * This effectively removes the oldest signature from the buffer.
+ */
+static void
+shift_signature_data_buffer(sv_unthreaded_plugin_t *self)
+{
+  const int idx = self->signature_data_buffer_idx;
+  for (int j = 0; j < idx; j++) {
+    self->signature_data_buffer[j] = self->signature_data_buffer[j + 1];
+  }
+
+  self->signature_data_buffer_idx -= 1;
+  if (self->signature_data_buffer_idx < 1) self->signature_generated = false;
```

Comment:
I can't really see how this helps. `signature_generated` is not really used anymore.